src/coreclr/gc/gc.cpp

```diff
@@ -24943,8 +25017,17 @@ void gc_heap::process_last_np_surv_region (generation* consing_gen,
                 {
                     assert (next_plan_gen_num == 0);
                     next_region = get_new_region (0);
-                    dprintf (REGIONS_LOG, ("h%d getting a new region for gen0 plan start seg to %Ix",
-                        heap_number, heap_segment_mem (next_region)));
+                    if (next_region)
+                    {
+                        dprintf (REGIONS_LOG, ("h%d getting a new region for gen0 plan start seg to %Ix",
+                            heap_number, heap_segment_mem (next_region)));
+                    }
+                    else
+                    {
+                        dprintf (REGIONS_LOG, ("h%d couldn't get a region to plan gen0, special sweep on",
+                            heap_number));
+                        special_sweep_p = true;
```

Comment:
I think it would be good to put a comment explaining how the special sweep mode works, i.e. what's special about it.